xbmc/pvr/PVRDatabase.cpp

```diff
@@ -113,17 +113,21 @@ void CPVRDatabase::CreateTables()
       ")"
   );
 
+  bool bisMySQL = StringUtils::EqualsNoCase(
+      CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_databaseTV.type, "mysql");
+
   CLog::LogFC(LOGDEBUG, LOGPVR, "Creating table 'channelgroups'");
-  m_pDS->exec("CREATE TABLE channelgroups ("
-              "idGroup         integer primary key,"
-              "bIsRadio        bool, "
-              "iGroupType      integer, "
-              "sName           varchar(64), "
-              "iLastWatched    integer, "
-              "bIsHidden       bool, "
-              "iPosition       integer, "
-              "iLastOpened     integer"
-              ")");
+  m_pDS->exec(PrepareSQL("CREATE TABLE channelgroups ("
+                         "idGroup         integer primary key,"
+                         "bIsRadio        bool, "
+                         "iGroupType      integer, "
+                         "sName           varchar(64), "
+                         "iLastWatched    integer, "
+                         "bIsHidden       bool, "
+                         "iPosition       integer, "
+                         "iLastOpened     %s"
+                         ")",
+                         !bisMySQL ? "integer" : "bigint unsigned"));
```

Comment:
Why do we need bigint unsigned? Surely bigint is sufficent?

Also, I think bigint is also a valid type for sqlite so no need to embed the condition. It will just map it internally to integer.

Reply:
when i use BIGINT 'only' my setup with pvr.hts throws:
```
ERROR <general>: ExecuteQuery - failed to execute query 'UPDATE channelgroups SET iLastOpened =
   18446744073067449536 WHERE idGroup = 1'
ERROR <general>: SQL: [TV38] Undefined MySQL error: Code (1264)
   Query: UPDATE channelgroups SET iLastOpened = 18446744073067449536 WHERE idGroup = 1
```
that's why i chose BIGINT UNSIGNED

Reply:
But why is the number so large?

(Epoch time secs * 1000) should be the max and that number looks much bigger.

Reply:
The range of BIGINT UNSIGNED is 0 to 18446744073709551615, while BIGINT max is 922337203685477580, so to know if that is sufficient can you clarify what encoding is used for iLastWatched.

Yes because of type affliliation SQLite will accept BIGINT UNSIGNED too and maps in as dynamic type INTEGER (but you can't use it as a primary key). However if you choose to go that way then you will have to recreate the table on migration as SQLite does not support MODIFY COLUMN.

Reply:
i don't know, that's the test result with pvr.hts

Reply:
It's milliseconds since 1970 so it should be ok.

E.g. right is 1618561813000 which is far less than 922337203685477580.

Reply:
Since it is never going to be negative UNSIGNED makes sense to me. As 4 byte INT UNSIGNED with max 4294967295 is not sufficient, go with BIGINT UNSIGNED

Reply:
ok, then we need to go here with `!bisMySQL ? "UNSIGNED BIG INT" : "BIGINT UNSIGNED"));` according to https://sqlite.org/datatype3.html Section 3.1.1. Affinity Name Examples, right?

Reply:
No, I tested and SQLite can handle "BIGINT UNSIGNED", the documentation only gives some of the variations as examples.

Reply:
cool, thanks